### PR TITLE
Fix BitNav data binding (#477)

### LINKED ISSUE #477: Change selectedkey parameter after click on a link in BitNav component
The Selectedkey parameter should change to the key of current link and its style should be change to selected.

### PR DESCRIPTION
**Pull request by khashayar-pakkhesal**
this PR closes #477

### CODE REVIEW

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs [hunk 1]
```diff
@@ -1,16 +1,43 @@
...
     {
         /// <summary>
         /// The key of the nav item selected by caller
         /// </summary>
-        [Parameter] public string? SelectedKey { get; set; }
+        private string? selectedKey;
+
+        [Inject] private NavigationManager navigationManager { get; set; }
+
+        public bool SelectedKeyHasBeenSet { get; set; }
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs:19]: please change this public property to a private field

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
+
+        [Parameter]
+        public string SelectedKey
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs:22]: please change this property's type to "string?" in order to resolve the warnings

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs [hunk 3]
```diff
@@ -1,16 +1,43 @@
...
+        {
+            get => selectedKey;
+            set
+            {
+                if (value == selectedKey) return;
+                selectedKey = value;
+                SelectedKeyChanged.InvokeAsync(selectedKey);
+
+                var currentUrl = NavLinkItems.Where(nli => nli.Links != null).SelectMany(sli => sli.Links).FirstOrDefault(sli => sli.Key.Contains(selectedKey))?.Key ?? string.Empty;
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs:31]: please format this line into multiple lines

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs [hunk 4]
```diff
@@ -34,6 +61,29 @@ public partial class BitNav
...
+
+
+            await base.OnInitializedAsync();
+        }
+
+        private void OnLocationChanged(object sender, LocationChangedEventArgs args)
+        {
+            var currentPage = navigationManager.Uri.Replace(navigationManager.BaseUri, string.Empty);
+
+            string currentPageKey = NavLinkItems.Where(nli => nli.Links != null).SelectMany(nli => nli.Links).FirstOrDefault(l => l.Url.ToLower().Contains(currentPage.ToLower()))?.Key ?? string.Empty;
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs:78]: please format this line into multiple lines

**Review by msynk — CHANGES_REQUESTED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs (hunk 2 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs]: Changed to nullable string

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs (hunk 1 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs]: Changed to private

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs (hunk 4 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs]: formatted into multiple lines

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs (hunk 3 above)
Comment (khashayar-pakkhesal) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs]: formatted into multiple lines

**Review by msynk — APPROVED** [review]

**Pull request merged**